chainstate/tx-verifier/src/transaction_verifier/transferred_amount_check.rs

```diff
@@ -390,6 +432,72 @@ pub fn calculate_tokens_burned_in_outputs(
         .ok_or(ConnectTransactionError::BurnAmountSumError(tx.get_id()))
 }
 
+fn calculate_required_fee_burn(
```

Comment:
I don't mind these gymnastics, considering they're temporary... but wouldn't have been easier to just make a copy of the current input calculation method, and add these to it as subtraction to the total?

Reply:
Not sure it's rather big, I'd keep it like this